### PR TITLE
Bugfix for experiment data export storing condition name in condition payload column

### PR DESCRIPTION
**Pull request by ppratikcr7**
This resolves ticket #2090

### CODE REVIEW

**Review by VivekFitkariwala — CHANGES_REQUESTED** [review]
Can you add proper types in the relevant code? This error came because in
`let experiment = acc.find((e) => e.experimentId === item.experimentId);` experiment is `any`

`let experiment` is `ExperimentCSVData` and the same variable is used for the experiment.

**Comment by danoswaltCL** [comment]
@amurphy-cl should this go in the release?

**Review by VivekFitkariwala — CHANGES_REQUESTED** [review]

**Comment by danoswaltCL** [comment]
I believe that @amurphy-cl said go ahead and put this into the release

File: backend/packages/Upgrade/src/api/services/AnalyticsService.ts [hunk 1]
```diff
@@ -183,7 +183,7 @@ export class AnalyticsService {
         throw new HttpError(404, `Experiment not found for id: ${experimentId}`);
       }
       const formattedExperiments = experimentQueryResult.reduce((acc, item) => {
```
[the remark this answers is not in the document]
Comment (ppratikcr7) [reply backend/packages/Upgrade/src/api/services/AnalyticsService.ts:185]: Yes, I will update the type to `ExperimentCSVData[]`. So, `experimentQueryResult` has the data row for each condition payload. We are getting the experiment generic data first and then we add the payload details in a separate key called `details`. Later we use this `formattedExperiments` to get the individual enrolment data for that experiment. So, its basically we are not checking for unique values but pushing all the condition payload details in the `formattedExperiments`. The naming is nit odd, will update them.

**Comment by ppratikcr7** [comment]
Closing this as its handled by a new [PR](https://github.com/CarnegieLearningWeb/UpGrade/pull/2095)

**Pull request closed without merging**